tensorflow/core/util/gpu_launch_config.h

```diff
@@ -424,7 +424,18 @@ Status CudaLaunchKernel(void (*function)(Ts...), dim3 grid_dim, dim3 block_dim,
 }
 #endif  // GOOGLE_CUDA
 
+#if GOOGLE_CUDA
+#define GpuLaunchKernel(kernel, block_count, threads_per_block, shared_mem,   \
```

Comment:
@jerryyin I think what we want to do is to:

- make `GpuLaunchKernel` a function, not a macro
- make an alias of `CudaLaunchKernel` based on the infrastructure you built #28343
- retain the call site of ops be `TF_RET_CHECK(CudaLaunchKernel` to minimize impacts to upstream logic

Reply:
+1

Reply:
also break the single commit into separate commits
1. commit to introduce the `GpuLaunchKernelFunction` (note we will also need to continue to have `CudaLaunchKernel` for "backwards compatibility"). We need to get this one right first
2. commit to revert `GPU_LAUNCH_KERNEL` call sites back to `TF_CHECK_OK(GpuLaunchKernel`
3. commit to remove the `GPU_LAUNCH_KERNEL` definition

Note that this breakdown is just for ease of review...you can squash them into one, when applying the same to PR 28343

Reply:
Thanks guys. Well explained. I'll reset the head and start from the first commit.

Reply:
Per our latest discussion. I will finish this PR until amend the upstream PR 28343.

When amending the upstream PR 28343, I will cherry pick commit 1 and commit 3 to upstream. From @Deven-amd's suggestion. We will hold off commit 2 as a large PR altogether in the end.